### ISSUE TITLE
When `markdown-fontify-whole-heading-line` is enabled, the leading `#` characters in headers are not fontified the same as the heading

### ISSUE THREAD

**Issue by jamescherti** [jrblevin/markdown-mode#848]
## Expected Behavior

The `defcustom` option `markdown-fontify-whole-heading-line` is intended to apply fontification to the entire line for headings.

## Actual Behavior

When `markdown-fontify-whole-heading-line` is set to `t`, the leading `#` characters are not fontified with the same face as the headings; they retain the `markdown-header-delimiter-face` instead.

## Steps to Reproduce

1. Set `markdown-fontify-whole-heading-line` to `t`
2. Open a Markdown file

## Backtrace

None

## Software Versions

Emacs 30.0.92

**Comment by jamescherti** [comment]
This bug was introduced in the following commit:
```
commit ece4f28a3de756e85240a53ccbbcffa7f49a06e2 (HEAD)
Author: Tomohisa Kuranari <tomohisa.kuranari@gmail.com>
Date:   Sat Jul 22 15:51:11 2023 +0900

    Hide markup even if markdown-fontify-whole-heading-line is enabled
```

The commit prior to the one above is working:
```
commit 2e2108c34cdcc8e58e00ceb3d8305a3813dac04c (HEAD)
Author: Shohei YOSHIDA <syohex@gmail.com>
Date:   Fri Jul 21 10:46:34 2023 +0900

    Add issue URLs
```

**Comment by jamescherti** [comment]
Pull request to address this issue: https://github.com/jrblevin/markdown-mode/pull/850

**Comment by syohex** [comment]
Merged #850
